### PR TITLE
[AzCopyV10][Feature] Display more fields for list command

### PR DESCRIPTION
**Pull request by mohsha-msft**
Added flag `--properties` in the list command which takes `;` separated list of properties and will display these fields in list command output. 
Currently list of supported properties is `"LastModifiedTime", "VersionId", "BlobType", "BlobAccessTier", "ContentType", and "ContentEncoding"` . Based on requirement, we can add more parameters in the future.

### CODE REVIEW

File: cmd/list.go [hunk 1]
```diff
@@ -178,17 +255,17 @@ func HandleListContainerCommand(unparsedSource string, location common.Location)
...
+//func printListContainerResponse(lsResponse *common.ListContainerResponse) {
+//	if len(lsResponse.Blobs) == 0 {
+//		return
+//	}
+//	// TODO determine what's the best way to display the blobs in JSON
+//	// TODO no partner team needs this functionality right now so the blobs are just outputted as info
+//	for index := 0; index < len(lsResponse.Blobs); index++ {
+//		glcm.Info(lsResponse.Blobs[index])
+//	}
+//}
```
Comment (zezha-msft) [inline cmd/list.go:268]: we can remove this if it's not used.

File: cmd/list.go [hunk 2]
```diff
@@ -26,15 +26,80 @@ import (
...
+	// obtained from argument
+	sourcePath string
+
+	Properties      string
+	MachineReadable bool
+	RunningTally    bool
+	MegaUnits       bool
+}
+
+var validProperties = []string{
```
Comment (zezha-msft) [inline cmd/list.go:47]: perhaps you could consider using another data structure to make the comparison easier, and also ignore the casing.

**Comment by zezha-msft** [comment]
> 		Example: listCmdExample,

Update examples.

---
Refers to: cmd/list.go:110 in b73c6dc. [](commit_id = b73c6dcf77ac1f4372d0c84e5fbb76a359f72cf7, deletion_comment = False)

File: cmd/list.go [hunk 3]
```diff
@@ -26,15 +26,80 @@ import (
...
+	"LastModifiedTime", "VersionId", "BlobType", "BlobAccessTier", "ContentType", "ContentEncoding",
```
Comment (zezha-msft) [inline cmd/list.go:48]: >LastModifiedTime [](start = 2, length = 16)

Sorry I should have given you this info earlier, I found the original ask and it included lease status too.

Comment (mohsha-msft) [reply cmd/list.go]: No problem. Changes weren't that big. Added `LeaseStatus`, `LeaseState`, and `LeaseDuration` in the `storedObject` as well.

File: cmd/list.go (hunk 2 above)
Comment (mohsha-msft) [reply cmd/list.go]: I am using `strings.EqualFolds()` for string comparison so it is case-insensitive. As for the data-structure, I've converted it to `const` to keep the behavior similar as other similar types (`BlobType`, `BlobAccessTier`, etc.).

File: cmd/list.go [hunk 4]
```diff
@@ -208,13 +296,13 @@ func byteSizeToString(size int64) string {
 		"GiB",
 		"TiB",
 		"PiB",
-		"EiB", // Let's face it, a file, account, or container probably won't be more than 1000 exabytes in YEARS. (and int64 literally isn't large enough to handle too many exbibytes. 128 bit processors when)
+		"EiB", // Let's face it, a file, account, or container probably won't be more than 1000 exabytes in YEARS. (and int64 literally isn't large enough to handle too many exabytes. 128 bit processors when)
```
Comment (nakulkar-msft) [inline cmd/list.go:299]: Change this back to exbiytes. The correct term here is exbibytes.

**Review by nakulkar-msft — CHANGES_REQUESTED** [review]

File: cmd/list.go (hunk 4 above)
Comment (mohsha-msft) [reply cmd/list.go:299]: Sure.

**Review by nakulkar-msft — APPROVED** [review]

**Pull request merged**